hpoflow/optuna_transformers.py

```diff
@@ -0,0 +1,125 @@
+# Copyright (c) 2021 Timothy Wolff-Piggott
+# This software is distributed under the terms of the MIT license
+# which is available at https://opensource.org/licenses/MIT
+
+import logging
+import os
+
+import mlflow
+
+from hpoflow.optuna_mlflow import OptunaMLflow
+from optuna._imports import try_import
+
+with try_import() as _imports:
+    import transformers
+_imports.check()
+
+_logger = logging.getLogger(__name__)
+
+
+class OMLflowCallback(transformers.TrainerCallback):
+    """
+    Class based on ``transformers.TrainerCallback``; integrates with OptunaMLflow
+    to send the logs to ``MLflow`` and ``Optuna`` during model training.
+    """
+
+    def __init__(
+        self,
+        trial: OptunaMLflow,
+        log_training_args: bool = True,
+        log_model_config: bool = True,
+    ):
+        """
+        Check integration package dependencies and initialize class.
+        Args:
+            trial: OptunaMLflow object
+            log_training_args: Whether to log all Transformers TrainingArguments as MLflow params
+            log_model_config: Whether to log the Transformers model config as MLflow params
+        """
+        self._MAX_PARAM_VAL_LENGTH = mlflow.utils.validation.MAX_PARAM_VAL_LENGTH
+        self._MAX_PARAMS_TAGS_PER_BATCH = mlflow.utils.validation.MAX_PARAMS_TAGS_PER_BATCH
```

Comment:
Would it be better not to create these variables but use `mlflow.utils.validation.MAX_PARAM_VAL_LENGTH` and `mlflow.utils.validation.MAX_PARAMS_TAGS_PER_BATCH` directly below?

Reply:
I don't have a strong opinion either way. Happy to just access the variables directly.